core/chains/evm/config/toml/config.go

```diff
@@ -432,6 +434,57 @@ func (t *Transactions) setFrom(f *Transactions) {
 	if v := f.ResendAfterThreshold; v != nil {
 		t.ResendAfterThreshold = v
 	}
+	t.AutoPurge.setFrom(&f.AutoPurge)
+}
+
+type AutoPurgeConfig struct {
+	AutoPurgeStuckTxs        *bool
+	AutoPurgeThreshold       *uint32
+	AutoPurgeMinAttempts     *uint32
+	AutoPurgeDetectionApiUrl *commonconfig.URL
+}
+
+func (a *AutoPurgeConfig) ValidateConfig() (err error) {
+	if a.AutoPurgeStuckTxs == nil || !*a.AutoPurgeStuckTxs {
+		return
+	}
+
+	if a.AutoPurgeDetectionApiUrl != nil && !a.AutoPurgeDetectionApiUrl.IsZero() {
+		switch a.AutoPurgeDetectionApiUrl.Scheme {
+		case "http", "https":
+			return
+		default:
+			return commonconfig.ErrInvalid{Name: "AutoPurgeDetectionApiUrl", Value: a.AutoPurgeDetectionApiUrl.Scheme, Msg: "must be http or https"}
+		}
+	}
+
+	if a.AutoPurgeThreshold == nil {
+		err = errors.Join(err, commonconfig.ErrMissing{Name: "AutoPurgeThreshold", Msg: "needs to be set if AutoPurgeStuckTxs is enabled and AutoPurgeDetectionApiUrl not set"})
+	} else if *a.AutoPurgeThreshold == 0 {
+		err = errors.Join(err, commonconfig.ErrInvalid{Name: "AutoPurgeThreshold", Value: *a.AutoPurgeThreshold, Msg: "cannot be 0 if AutoPurgeStuckTxs is enabled and AutoPurgeDetectionApiUrl not set"})
+	}
+	if a.AutoPurgeMinAttempts == nil {
```

Comment:
Should we also validate here that gas bumping is enabled?

Reply:
I think we should for robustness but it could get messy with how we scope these configs. This one being under `Transactions` but that one being under `GasEstimators=`

Reply:
I moved the config validations under `Chain`'s so I could add the `BumpThreshold` check. I also made the validations chain specific since they require different configs based on their detection method.